stdlib/messaging/activemq-artemis/src/main/java/org/ballerinalang/messaging/artemis/externimpl/message/GetBodySize.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.ballerinalang.messaging.artemis.externimpl.message;
+
+import org.apache.activemq.artemis.api.core.client.ClientMessage;
+import org.ballerinalang.bre.Context;
+import org.ballerinalang.bre.bvm.BlockingNativeCallableUnit;
+import org.ballerinalang.messaging.artemis.ArtemisConstants;
+import org.ballerinalang.messaging.artemis.ArtemisUtils;
+import org.ballerinalang.model.types.TypeKind;
+import org.ballerinalang.model.values.BInteger;
+import org.ballerinalang.model.values.BMap;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.Receiver;
+
+/**
+ * Extern function to get the body size of an artemis message.
+ *
+ * @since 0.995
+ */
+
+@BallerinaFunction(
+        orgName = ArtemisConstants.BALLERINA, packageName = ArtemisConstants.ARTEMIS,
+        functionName = "getBodySize",
+        receiver = @Receiver(type = TypeKind.OBJECT, structType = ArtemisConstants.MESSAGE_OBJ,
+                             structPackage = ArtemisConstants.PROTOCOL_PACKAGE_ARTEMIS),
+        isPublic = true
+)
+public class GetBodySize extends BlockingNativeCallableUnit {
+
+    @Override
+    public void execute(Context context) {
+        try {
+            @SuppressWarnings(ArtemisConstants.UNCHECKED)
+            BMap<String, BValue> messageObj = (BMap<String, BValue>) context.getRefArgument(0);
+            ClientMessage message = (ClientMessage) messageObj.getNativeData(ArtemisConstants.ARTEMIS_MESSAGE);
+            context.setReturnValues(new BInteger(message.getBodySize()));
+        } catch (Exception e) {
+            context.setReturnValues(ArtemisUtils.getError(context, e));
```

Comment:
Why we can't handle all the specific exceptions here instead of this? Check the other places as well.

Reply:
Since this method returns an error we do not want this method to throw an exception and fail. Instead we make sure it returns all errors. This is the way most of the Ballerina functions are written including http ones. 
Eg:
https://github.com/ballerina-platform/ballerina-lang/blob/019bb4a98312a3a8eb860b9374f672dd6fca2023/stdlib/mime/src/main/java/org/ballerinalang/mime/nativeimpl/GetBodyAsString.java#L68-L71
https://github.com/ballerina-platform/ballerina-lang/blob/019bb4a98312a3a8eb860b9374f672dd6fca2023/stdlib/mime/src/main/java/org/ballerinalang/mime/nativeimpl/GetByteArray.java#L92-L95

Reply:
We need to come to a final decision about this and apply to all the places.